src/parse.ts

```diff
@@ -98,6 +108,25 @@ export function fixupTypes(types: MessageDefinition[]): void {
   });
 }
 
+function buildRos2IdlType(messageDefinition: string, grammar: Grammar): MessageDefinition[] {
+  const parser = new Parser(grammar);
+  parser.feed(messageDefinition);
+  const results = parser.finish();
+
+  if (results.length === 0) {
+    throw new Error(
+      `Could not parse message definition (unexpected end of input): '${messageDefinition}'`,
+    );
+  }
+  const result = results[0] as MessageDefinition[];
```

Comment:
Why the cast here? Are we not getting the correct type information from `results`? Is there a way to get correct type information?

Reply:
the parser has no knowledge of the output type. This is the same thing we're doing in `buildType` previously